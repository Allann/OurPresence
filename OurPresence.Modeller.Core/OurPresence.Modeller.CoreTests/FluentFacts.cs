// Copyright (c)  Allan Nielsen.
// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.

using OurPresence.Modeller.Domain;
using OurPresence.Modeller.Domain.Extensions;
using Xunit;
using ApprovalTests;

namespace OurPresence.Modeller.CoreTests
{
    public class FluentFacts
    {
        [Fact]
        public void Fluent_Module_CreateSucceeds()
        {
            var module = Fluent.Module
                .Create("Company","Project")
                .DefaultSchema("dbo")
                .FeatureName("Feature")
                .AddModel("Model")
                    .IsAuditable(true)
                    .Schema("dbo")
                    .WithDefaultKey()
                    .AddField("Field1")
                        .BusinessKey(true)
                        .DataType(DataTypes.String)
                        .Default("Default")
                        .MaxLength(10)
                        .MinLength(5)
                        .Build
                    .AddField("Field2")
                        .DataType(DataTypes.Decimal)
                        .Scale(12)
                        .Precision(2)
                        .Nullable(true)
                        .MinLength(200)
                        .MaxLength(3000)
                        .Build
                    .Build
                .AddEnumeration("Enumeration")
                    .AsFlag()
                    .AddItem("None")
                    .AddItem("One")
                    .AddItem("Two")
                    .AddItem("Three")
                    .Build
                .AddRequest("ChangeModel")
                    .AddField("ModelId")
                        .Build
                        .WithResponse()
                            .AddField("ModelName")
                            .Build
                        .Build
                    .Build
                .Build;

            Approvals.Verify(module.ToJson());
        }
    }
}
