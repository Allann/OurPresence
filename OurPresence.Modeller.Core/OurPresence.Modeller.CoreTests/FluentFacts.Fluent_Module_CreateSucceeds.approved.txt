{"company":"Company","project":"Project","feature":"Feature","defaultSchema":"dbo","models":[{"schema":"dbo","key":{"fields":[{"dataType":"UniqueIdentifier","name":"Id"}]},"fields":[{"default":"Default","maxLength":10,"minLength":5,"businessKey":true,"name":"Field1"},{"maxLength":3000,"minLength":200,"precision":2,"scale":12,"dataType":"Decimal","nullable":true,"name":"Field2"}],"indexes":[],"relationships":[],"behaviours":[],"name":"Model"}],"enumerations":[{"flag":true,"items":[{"name":"None","display":"None","value":0},{"name":"One","display":"One","value":1},{"name":"Two","display":"Two","value":2},{"name":"Three","display":"Three","value":4}],"name":"Enumeration"}],"requests":[{"response":{"fields":[{"name":"ModelName"}]},"fields":[{"name":"ModelId"}],"name":"ChangeModelQuery"}]}